File: wiki_java/src/main/java/com/github/klee0kai/wiki/provide/binding/SunSystemComponent.java
package com.github.klee0kai.wiki.provide.binding;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.stone.annotations.module.BindInstance;
import com.github.klee0kai.test.mowgli.galaxy.Sun;

@Component
public interface SunSystemComponent {

    @BindInstance
    Sun sun(Sun sun);

}

File: wiki_kotlin/src/main/kotlin/com/github/klee0kai/wiki/provide/binding/SunSystemComponent.kt
package com.github.klee0kai.wiki.provide.binding

import com.github.klee0kai.stone.annotations.component.Component
import com.github.klee0kai.stone.annotations.module.BindInstance
import com.github.klee0kai.test.mowgli.galaxy.Sun

@Component
interface SunSystemComponent {

    @BindInstance
    fun sun(sun: Sun? = null): Sun

}
